vault/src/robocorp/vault/__init__.py

```diff
@@ -79,91 +16,91 @@ def _get_vault():
-        i.e.: For each value, `log.hide_from_output(value)` will be called.
+        If `robocorp.log` is available in the environment, the `hide` argument
+        controls if the given values are automatically hidden in the log
+        output.
 
     Returns:
         The container for the secret (which has key-value pairs).
 
     Raises:
         RobocorpVaultError: Error with API request or response payload.
     """
-    try:
-        from robocorp import log  # type: ignore [attr-defined]
-
-        ctx = log.suppress_variables()
-        has_log = True
-    except ImportError:
-        pass  # If robocorp.log is not being used that's OK.
-        ctx = nullcontext()
-        has_log = False
-
-    with ctx:
+    with _suppress_variables():
         vault = _get_vault()
+        secret = vault.get_secret(name)
 
-        # Note: secrets are always gotten from the backend when requested
-        # (so any updates will be gotten in a new `get_secret` call).
-        secret = vault.get_secret(secret_name)
-
-        if has_log:
-            # There's no caching in place, so, on any new call it's possible
-            # that values changed, so, call this every time.
-            _ignore_secret_values(secret)
+        if hide:
+            _hide_secret_values(secret)
 
-    return secret
+        return secret
 
 
-def set_secret(secret: SecretContainer) -> None:
-    """Overwrite an existing secret with new values.
+def set_secret(
+    name: str,
+    values: dict[str, Any],
+    description: str = "",
+    exist_ok: bool = False,
+    hide: bool = True,
+) -> None:
+    """Create a new secret, or overwrite an existing one.
 
-    Note:
-        Only allows modifying existing secrets, and replaces
-          all values contained within it.
+    Args:
+        name: Name of secret
+        values: Mapping of secret keys and values
+        description: Optional description for secret
+        exist_ok: Overwrite existing secret
+        hide: Hide secret values from log output
 
     Note:
-        When used, if `robocorp.log` is in the environment, all the values
-        set will be automatically hidden from the logs.
-        i.e.: For each value, `log.hide_from_output(value)` will be called.
+        If `robocorp.log` is available in the environment, the `hide` argument
+        controls if the given values are automatically hidden in the log
+        output.
 
-    Args:
-        secret: the secret object which was mutated.
+    Raises:
+        RobocorpVaultError: Error with API request or response payload
     """
-    try:
-        from robocorp import log  # type: ignore [attr-defined]
+    with _suppress_variables():
+        vault = _get_vault()
 
-        ctx = log.suppress_variables()
-        has_log = True
-    except ImportError:
-        pass  # If robocorp.log is not being used that's OK.
-        ctx = nullcontext()
-        has_log = False
+        try:
+            vault.get_secret(name)
```

Comment:
If `exist_ok` is True, getting the secret first shouldn't be needed (so, one less network request).